include/boost/geometry/strategies/cartesian/side_rounded_input.hpp

```diff
@@ -0,0 +1,60 @@
+// Boost.Geometry (aka GGL, Generic Geometry Library)
+
+// Copyright (c) 2021 Tinko Bartels, Berlin, Germany.
+
+// Use, modification and distribution is subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_GEOMETRY_STRATEGY_CARTESIAN_SIDE_ROUNDED_INPUT_HPP
+#define BOOST_GEOMETRY_STRATEGY_CARTESIAN_SIDE_ROUNDED_INPUT_HPP
+
+#include <boost/geometry/core/config.hpp>
+
+#include <boost/geometry/strategies/side.hpp>
+
+
+namespace boost { namespace geometry
+{
+
+namespace strategy { namespace side
+{
+
+
+template <typename CalculationType>
+struct side_rounded_input
+{
+    using cs_tag = cartesian_tag;
+    template <typename P1, typename P2, typename P>
+    static inline int apply(P1 const& p1, P2 const& p2, P const& p)
+    {
+
+        using CT = std::conditional_t
+            <
+                std::is_void<CalculationType>::value,
+                double,
+                CalculationType
+            >;
```

Comment:
Typically if `CT` is `void` we use the most precise coordinate type. Or is there a reason why `double` is used here?